### DISCUSSION TITLE
New alignment / justification icons

### DISCUSSION THREAD

**Discussion by jameskoster** [WordPress/gutenberg#47153] [Show and tell]
Work under way in https://github.com/WordPress/gutenberg/pull/47134 will add some new alignment and justification options for blocks, specifically `align-items: stretch` and `justify-content: space-between` equivalents.

We don't currently have dedicated icons for this purpose, and so should expand the set. The problem is that the design convention we've used for alignment icons so far makes use of a single item to represent the object being aligned, and it is quite tricky to represent space-between (and stretching to a lesser extent) with a single item:

<img width="273" alt="Screenshot 2023-01-13 at 17 12 37" src="https://user-images.githubusercontent.com/846565/212378871-0f2826d4-a0fb-440b-bcf6-f5bc350bd778.png">

Let's explore some alternative conventions.

**Comment by jameskoster** [comment]
Here's an option that uses two items instead of one, which makes it much easier to communicate `space-between` and `stretch`. There's a version with 'guide' lines and one without:

<img width="616" alt="Screenshot 2023-01-13 at 17 14 04" src="https://user-images.githubusercontent.com/846565/212379134-cf3d1164-695c-423c-bbd8-51ef2f0cf57c.png">

cc @WordPress/gutenberg-design 

These are in the [WordPress Design Library Figma file](https://www.figma.com/file/e4tLacmlPuZV47l7901FEs/WordPress-Design-Library?node-id=52420%3A15395&t=waaWXr2BfJ7EVvfU-1) in case anyone would like to riff on them.

**Reply by jasmussen** [reply]
I think option 1, or a variation of option 1 could work. Option 2 becomes a bit more noisy with 3 elements per design. But it'd need a bit of finessing the curves and spacing — perhaps it's the aliasing of the exported PNG, but it feels like there's some airiness and crispness to the existing icons that gets a bit lost in the new.

**Reply by jameskoster** [reply]
I see @richtabor added that to the Figma:

<img width="297" alt="Screenshot 2023-01-16 at 13 01 30" src="https://user-images.githubusercontent.com/846565/212684301-e64a565e-c78b-4684-8fa4-e6e9f2644571.png">

It works for me.

**Reply by jameskoster** [reply]
Option 4: Fewer guide lines (I think we probably need to keep them for `stretch` and `space-between`):

<img width="288" alt="Screenshot 2023-01-16 at 13 05 42" src="https://user-images.githubusercontent.com/846565/212685082-dee50fa4-52f6-4697-b22c-92dc71546c04.png">

**Reply by jasmussen** [reply]
These are getting closer. They still seem a bit big, and it might be good to see them in context of the block toolbar 🤔

**Reply by jasmussen** [reply]
I think these work well alongside the existing ones:

<img width="246" alt="Screenshot 2023-01-16 at 14 30 39" src="https://user-images.githubusercontent.com/1204802/212689905-9bd9c9eb-8d34-4e50-af65-76e74c72bb15.png">

Can we try these with 1 cell between the lines, instead of 2?
<img width="204" alt="Screenshot 2023-01-16 at 14 30 46" src="https://user-images.githubusercontent.com/1204802/212689989-a9206f8d-dbc7-4279-977b-7ecbd3124dad.png">

**Reply by jameskoster** [reply]
One cell can probably work in the mean time:

<img width="238" alt="Screenshot 2023-01-16 at 13 34 55" src="https://user-images.githubusercontent.com/846565/212690690-1f6e3bd4-333a-4d2c-8113-5196c0248e0e.png">

**Reply by jasmussen** [reply]
Yep, that seems a good interim.
